### PR TITLE
refactor(compiler): add support for sanitizing properties and attributes

### PR DESCRIPTION
**Pull request by mmalerba**
Sets sanitizer functions when attempting to set sensitive properties and attributes

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/main/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?

sanitizer functions are not supported in template pipeline

Issue Number: N/A

## What is the new behavior? 
sanitizer functions are supported in template pipeline

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: packages/compiler/src/template/pipeline/src/phases/resolve_sanitizers.ts [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+          }
+          break;
+        case ir.OpKind.StyleProp:
+        case ir.OpKind.StyleMap:
+        case ir.OpKind.InterpolateStyleProp:
+        case ir.OpKind.InterpolateStyleMap:
+          // The compiler does not fill in a sanitizer for style binding values because the
+          // style algorithm knows internally what props are subject to sanitization (only style
+          // values set via [attr.style] are explicitly sanitized)
+          // TODO: is the context ever anything other than style for these props? if not we can just
```
Comment (mmalerba) [inline packages/compiler/src/template/pipeline/src/phases/resolve_sanitizers.ts:51]: Can we just completely remove sanitization for style prop/map?

File: packages/compiler/src/template/pipeline/src/phases/reify.ts [hunk 2]
```diff
@@ -7,11 +7,21 @@
  */
 
 import * as o from '../../../../output/output_ast';
+import {Identifiers} from '../../../../render3/r3_identifiers';
 import * as ir from '../../ir';
 
 import type {ComponentCompilation, ViewCompilation} from '../compilation';
 import * as ng from '../instruction';
 
+// TODO: Ok to pull in `Identifiers` here? I mostly see it used in `instruction.ts`
```
[the remark this answers is not in the document]
Comment (dylhunn) [reply packages/compiler/src/template/pipeline/src/phases/reify.ts:16]: I don't see why not? If there's a problem, @alxhub might know

File: packages/compiler/src/template/pipeline/src/phases/resolve_sanitizers.ts [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+          break;
+      }
+    }
+  }
+}
+
+/**
+ * Gets a map of all elements in the givne view by their xref id.
+ */
+// TODO: Is there a shared location I can put this? Copied the code from attribute_extraction.ts
```
[the remark this answers is not in the document]
Comment (dylhunn) [reply packages/compiler/src/template/pipeline/src/phases/resolve_sanitizers.ts:66]: Not yet, but feel free to create a shared.ts!

**Review by dylhunn — APPROVED** [review]
reviewed-for: fw-compiler

File: packages/compiler/src/template/pipeline/ir/src/ops/update.ts [hunk 4]
```diff
@@ -6,6 +6,15 @@
...
  */
 
+
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
```
Comment (alfaproject) [inline packages/compiler/src/template/pipeline/ir/src/ops/update.ts:16]: This header is duplicated now

**Comment by mmalerba** [comment]
Does not affect internal tests

**Comment by alxhub** [comment]
This PR was merged into the repository by commit c1052cf7a77e0bf2a4ec14f9dd5abc92034cfd2e.

**Pull request closed without merging**